src/Microsoft.Azure.Batch.SoftwareEntitlement/NodeEntitlementReader.cs

```diff
@@ -132,10 +132,22 @@ void ReadClaim(string claimId, Func<NodeEntitlements, string, NodeEntitlements>
 
             ReadClaim(Claims.VirtualMachineId, (e, val) => e.WithVirtualMachineId(val));
             ReadClaim(Claims.EntitlementId, (e, val) => e.WithIdentifier(val));
+            ReadClaim(Claims.CpuCoreCount, SetCpuCoreCount);
+            ReadClaim(Claims.BatchAccountId, (e, val) => e.WithBatchAccountId(val));
+            ReadClaim(Claims.PoolId, (e, val) => e.WithPoolId(val));
+            ReadClaim(Claims.JobId, (e, val) => e.WithJobId(val));
+            ReadClaim(Claims.TaskId, (e, val) => e.WithTaskId(val));
 
             return Errorable.Success(result);
         }
 
+        private static NodeEntitlements SetCpuCoreCount(NodeEntitlements entitlement, string value)
+        {
+            return int.TryParse(value, out int cpuCoreCount)
+                ? entitlement.WithCpuCoreCount(cpuCoreCount)
+                : entitlement;
```

Comment:
This silently discards any error - if `value` can't be parsed, we should generate an error.

Reply:
You could modify `ReadClaim` to take care of the type conversion:

    void ReadClaim<T>(string claimId, Func<NodeEntitlements, T, NodeEntitlements> action) { ... }

You'd need a conversion method like this one:
http://www.nichesoftware.co.nz/2014/08/24/easy-string-conversion.html

But then `ReadClaim` would reusably take care of any errors.

Reply:
I'll look into changing that...

Reply:
I'm getting a bit tangled up catching exceptions and returning `Errorables` with this approach.

For now, I'd prefer to change it to a simpler (but more verbose) version:
```
            var cpuCoreCountClaim = principal.FindFirst(Claims.CpuCoreCount);
            if (cpuCoreCountClaim != null)
            {
                if (int.TryParse(cpuCoreCountClaim.Value, out int cpuCoreCount))
                {
                    result = result.WithCpuCoreCount(cpuCoreCount);
                }
                else
                {
                    return Errorable.Failure<NodeEntitlements>(
                        InvalidTokenError($"Invalid CPU core count claim: {cpuCoreCountClaim.Value}"));
                }
            }
```

I think there are a few options for cleaning it up, and in the interests of keeping this PR as light as possible, it'd be good to explore those afterwards. Ideally, I'd like a solution that highlights possible conversion errors at compile time and also deals with collections of claims (e.g. IP addresses and application IDs). Something I was playing around with earlier (which _did_ have the proper error-handling semantics for CPU core count) looked like this:
```
return Builder<NodeEntitlements>.Create(principal, result)
	.WithMultipleClaims(Claims.Application, (e, vals) => e.WithApplications(vals))
	.WithMultipleClaims(Claims.IpAddress, ParseIpAddress, (e, vals) => e.WithIpAddresses(vals))
	.WithOptionalClaim(Claims.VirtualMachineId, (e, val) => e.WithVirtualMachineId(val))
	.WithOptionalClaim(Claims.CpuCoreCount, ParseCpuCoreCount, (e, val) => e.WithCpuCoreCount(val))
	.WithOptionalClaim(Claims.EntitlementId, (e, val) => e.WithIdentifier(val))
	.WithOptionalClaim(Claims.BatchAccountId, (e, val) => e.WithBatchAccountId(val))
	.WithOptionalClaim(Claims.PoolId, (e, val) => e.WithPoolId(val))
	.WithOptionalClaim(Claims.JobId, (e, val) => e.WithJobId(val))
	.WithOptionalClaim(Claims.TaskId, (e, val) => e.WithTaskId(val))
	.Result;
```
Think the naming needs some work, but the general approach may have potential.